api/client/system/events.go

```diff
@@ -70,26 +81,42 @@ func runEvents(dockerCli *client.DockerCli, opts *eventsOptions) error {
 	}
 	defer responseBody.Close()
 
-	return streamEvents(responseBody, dockerCli.Out())
+	return streamEvents(dockerCli, responseBody, tmpl)
+}
+
+func makeTemplate(format string) (*template.Template, error) {
+	if format == "" {
+		return nil, nil
+	}
+	tmpl, err := templates.Parse(format)
+	if err != nil {
+		return tmpl, err
+	}
+	// we execute the template for an empty message, so as to validate
+	// a bad template like "{{.badFieldString}}"
+	return tmpl, tmpl.Execute(ioutil.Discard, &eventtypes.Message{})
 }
 
 // streamEvents decodes prints the incoming events in the provided output.
-func streamEvents(input io.Reader, output io.Writer) error {
+func streamEvents(dockerCli *client.DockerCli, input io.Reader, tmpl *template.Template) error {
 	return DecodeEvents(input, func(event eventtypes.Message, err error) error {
 		if err != nil {
 			return err
 		}
-		printOutput(event, output)
-		return nil
+		if tmpl == nil {
+			return prettyPrintEvent(dockerCli, event)
+		}
+		return formatEvent(dockerCli, event, tmpl)
 	})
 }
 
 type eventProcessor func(event eventtypes.Message, err error) error
 
-// printOutput prints all types of event information.
+// prettyPrintEvent prints all types of event information.
 // Each output includes the event type, actor id, name and action.
 // Actor attributes are printed at the end if the actor has any.
-func printOutput(event eventtypes.Message, output io.Writer) {
+func prettyPrintEvent(dockerCli *client.DockerCli, event eventtypes.Message) error {
+	output := dockerCli.Out()
```

Comment:
I don't think this is correct. If the function only uses `Out()`, it should get passed just the `io.Writer`, not the entire `DockerCli` object. It will make it easier to refactor things later by passing just the dependencies to the function.